File: Java/357.java
static final int MAX = 100005;
static ArrayList<Integer> g[] =  new ArrayList[MAX]; 
static long dist[] = new long[MAX];
static int N, M; 

void bfs(int u) {
    Queue<Integer> q = new LinkedList<>();
    q.add(u);
    dist[u] = 0;

    while (!q.isEmpty()) {
        u = q.poll();
        for (int v : g[u]) {
            if (dist[v] == -1) {
                dist[v] = dist[u] + 1;
                q.add(v);
            }
        }
    }
}

static void init() {
    for(int i = 0; i <= N; i++) {
        g[i] = new ArrayList<>();
        dist[i] = -1;
    }
}

File: CPP/357.cpp
const int MX = 1e5+5;
vector<int> g[MX]; 
vector<ll> dist;
int n, m;

void bfs(int u) {
    queue<int> q;
    q.push(u);
    dist[u] = 0;

    while (q.size()) {
        u = q.front();
        q.pop();
        for (auto &v : g[u]) {
            if (dist[v] == -1) {
                dist[v] = dist[u] + 1;
                q.push(v);
            }
        }
    }
}

void init() {
    dist.assign(n, -1);
    for (int i = 0; i <= n; i++) {
        g[i].clear();
    }
}